src/v/cloud_roles/signature.h

```diff
@@ -127,6 +134,17 @@ template<class Fn>
 time_source::time_source(Fn&& fn, int)
   : _gettime_fn(std::forward<Fn>(fn)) {}
 
+template<int result_len>
+ss::sstring time_source::format(const char* fmt) const {
+    std::array<char, result_len> out_str{};
```

Comment:
is there a need to expose the size of the buffer? given the small dimension and the private context, 
std::array<char, std::max(formatted_date_len, std::max(formatted_datetime_len, http_formatted_datetime_len))>

Reply:
Switched the time formatting to use fmt in: https://github.com/redpanda-data/redpanda/commit/ec594d8a360aecd2bb5ff4d14b3bf60cfcdbcfc1.